components/constellation/constellation.rs

```diff
@@ -1247,6 +1247,14 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
             FromScriptMsg::SetFullscreenState(state) => {
                 self.embedder_proxy.send(EmbedderMsg::SetFullscreenState(source_top_ctx_id, state));
             }
+            FromScriptMsg::ShowIME(kind) => {
+                debug!("constellation got ShowIME message");
+                self.embedder_proxy.send(EmbedderMsg::ShowIME(source_top_ctx_id, kind));
```

Comment:
I think I would rather have us talk to the embedder from the script thread instead of going through the constellation.

Reply:
I didn't know this was a thing since we don't do it for other cases like TitleChange.

Reply:
IRC Chat: https://mozilla.logbot.info/servo/20180421#c14649458-c14649484
TL;DR: We should avoid adding lots of cases where the constellation is only forwarding a message when the message could be sent directly. But, I would be okay if this were a followup that removed other instances of this.

Also in the case of `TitleChange`, the constellation is checking if the source pipeline was the top-level pipeline.